CONTRIBUTING.md

```diff
@@ -160,6 +160,34 @@ Where again, `<PATTERN>` is the name of the folder listed in the
 [test/integration/tests/](https://github.com/commercialhaskell/stack/tree/master/test/integration/tests)
 folder.
 
+## CI Build rules
+
+We use [Azure](https://dev.azure.com/commercialhaskell/stack/_build)
+to do CI builds on Stack. There are two types of build which happens
+there:
+
+### Unit based build
+
+This build runs only the unit tests in the Stack codebase. These are
+the rules for triggering it:
+* CI will run this if commits are pushed to stable,master branch
+* CI will run this for any branches starting with `ci/`
+* CI will run this for all new PR's.
+
+### Integration based build
+
+This build runs the integration tests in the Stack codebase. This is
+scheduled to run daily once for both the stable and master branches.
+
+Also, you can manually run this on a specific branch from the Azure UI
+if you have the appropriate permissions.
```

Comment:
Perhaps comment: if you'd specifically like a branch or PR to run integration tests, add a comment in the PR and we can queue one up.

Reply:
Added them.